components/stepper/index.css

```
@@ -10,134 +10,560 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+                              + var(--mod-stepper-focus-indicator-width, var(--spectrum-stepper-focus-indicator-width)))
+                              * -1);
+  }
+
+  /*** Hover ***/
+  &:hover:not(.is-disabled):not(.is-invalid):not(.is-focused):not(.is-keyboardFocused) {
+    .spectrum-Stepper-buttons,
+    .spectrum-Stepper-stepUp,
+    .spectrum-Stepper-stepDown,
+    .spectrum-Stepper-input {
+      border-color: var(--highcontrast-stepper-border-color-hover, var(--mod-stepper-border-color-hover, var(--spectrum-stepper-border-color-hover)));
+    }
+  }
+
+
+  /*** Focused ***/
+  &.is-focused {
+    .spectrum-Stepper-input,
+    .spectrum-Stepper-buttons,
+    .spectrum-Stepper-stepUp,
+    .spectrum-Stepper-stepDown {
+      border-color: var(--highcontrast-stepper-border-color-focus, var(--mod-stepper-border-color-focus, var(--spectrum-stepper-border-color-focus)));
+    }
+
+    .spectrum-Stepper-stepUp,
+    .spectrum-Stepper-stepDown {
+      background-color: var(--highcontrast-stepper-button-background-color-focus, var(--mod-stepper-button-background-color-focus, var(--spectrum-stepper-button-background-color-focus)));
+    }
+
+    /* focused hover */
+    &:hover {
+      .spectrum-Stepper-input,
+      .spectrum-Stepper-buttons,
+      .spectrum-Stepper-stepUp,
+      .spectrum-Stepper-stepDown {
+        border-color: var(--highcontrast-stepper-border-color-focus-hover, var(--mod-stepper-border-color-focus-hover, var(--spectrum-stepper-border-color-focus-hover)));
+      }
+    }
+
+    /* focused invalid */
+    &.is-invalid {
+      .spectrum-Stepper-input,
+      .spectrum-Stepper-buttons,
+      .spectrum-Stepper-stepUp,
+      .spectrum-Stepper-stepDown {
+        border-color: var(--highcontrast-stepper-border-color-invalid-focus, var(--mod-stepper-border-color-invalid-focus, var(--spectrum-stepper-border-color-invalid-focus)));
+      }
+    }
+  }
+
+
+  /*** Keyboard Focused ***/
+  &.is-keyboardFocused {
+
+    /* keyboard focus indicator is visible */
+    &::after {
+      border-color: var(--highcontrast-stepper-focus-indicator-color, var(--mod-stepper-focus-indicator-color, var(--spectrum-stepper-focus-indicator-color)));
+      border-width: var(--mod-stepper-focus-indicator-width, var(--spectrum-stepper-focus-indicator-width));
+    }
+
+    .spectrum-Stepper-input,
+    .spectrum-Stepper-buttons,
+    .spectrum-Stepper-stepUp,
+    .spectrum-Stepper-stepDown {
+      border-color: var(--highcontrast-stepper-border-color-keyboard-focus, var(--mod-stepper-border-color-keyboard-focus, var(--spectrum-stepper-border-color-keyboard-focus)));
+    }
+
+    .spectrum-Stepper-stepUp,
+    .spectrum-Stepper-stepDown {
+      background-color: var(--highcontrast-stepper-button-background-color-keyboard-focus, var(--mod-stepper-button-background-color-keyboard-focus, var(--spectrum-stepper-button-background-color-keyboard-focus)));
+    }
+
+    /* keyboard focused invalid */
+    &.is-invalid {
```

```suggestion
    &.is-invalid,
    &:has(input:invalid) {
```

Another option we could leverage